packages/shared/src/components/ExtensionOnboarding.tsx

```diff
@@ -1,12 +1,22 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect } from 'react';
 import Link from 'next/link';
 import Logo, { LogoPosition } from './Logo';
 import { OnboardingTitleGradient } from './onboarding/common';
 import { onboardingUrl } from '../lib/constants';
 import { Button, ButtonSize } from './buttons/Button';
 import { cloudinary } from '../lib/image';
+import { useExtensionContext } from '../contexts/ExtensionContext';
 
 const ExtensionOnboarding = (): ReactElement => {
+  const { setCurrentPage } = useExtensionContext();
+
+  useEffect(() => {
+    setCurrentPage('/hijacking');
+    return () => {
+      setCurrentPage('/');
+    };
+  });
```

Comment:
I feel like `ExtensionOnboarding` should be rendered based on current page, not other way around where mounting of component triggers page change.

Reply:
Also using `useEffect` without deps array here would trigger it on each render

Reply:
I always mix them up, my idea was only trigger once, but that is empty deps array, not no-deps array.

Reply:
> I feel like `ExtensionOnboarding` should be rendered based on current page, not other way around where mounting of component triggers page change.

Not sure I get it. Set the current page, then render?

Reply:
Basically with this kind of approach when this component renders and sets the current page. Now if another component eg. `ExtensionPermissionsPrompt` renders after it, it will overwrite the current page as well. 

Now you can argue that they are never rendered at the same time but this kind of things get forgotten easily 😆 

So if we are going in this kind of direction where we are basically creating a little router for the extension I would rather have some other condition like `shouldRedirectOnboarding` actually update the `currentPage` and then in `App` we only render ExtensionOnboarding when `currentPage === '/onboarding`.

Reply:
Small issue with that, because of with FeaturesBoundary happening outside InternalApp, we would then not set currentPage to the right page for permissions.

Reply:
I think it works fine like this for now. If we end up adding more pages, we probably need to reconsider the router